NicolaeChristof/Rise
# Feature options menu

This PR adds the options menu, accessible from both the main menu and the pause menu. In the options menu, you can change the graphics level as well as the input method (controller or keyboard). The system also detects whether you have a controller plugged in so that you don't accidentally switch to controller when you have no controller keyboard.

Notes:
-Currently the options menu in the pause menu takes you back to the main menu overlay, but it doesn't reset your progress.

How to Test;
-Change settings in the options menu and ensure that they carry over to gameplay
-Test and make sure that change from controller to keyboard and vice versa doesn't mess up any multiplayer gameplay (I haven't been able to sufficiently test multiplayer)

EDIT:
I forgot to mention, but I also cleaned up the UI backend by including more clear comments on the UIController and separating the Canvas into 2 empty game objects: one for the HUD and one for the menus.
